src/sync/sync-utils.ts

```diff
@@ -49,6 +54,57 @@ export async function findOrStartSync(
 	}, 0, logger);
 }
 
+type subscriptionUpdateTasks = {
+	totalNumberOfRepos?: number | null;
+	repositoryCursor?: string | null;
+	repositoryStatus?: string | null;
+}
+
+const resetTargetedTasks = async (subscription: Subscription, syncType?: SyncType, targetTasks?: TaskType[]): Promise<void> => {
+	if (!syncType || !targetTasks?.length) {
+		return;
+	}
+
+	const subscriptionTasks = targetTasks.find(task => task === "repository");
+	const repoSyncTasks = targetTasks.filter(task => task !== "repository");
+	const updateRepoSyncTasks = {};
+	const updateSubscriptionTasks: subscriptionUpdateTasks = {};
```

Comment:
I'd probbly move this in the if statement below and start it with the default key `repositoryStatus: null`